React/forms.md

```diff
@@ -2,24 +2,266 @@
 
 ## Learning Goals
 - Implement a form in a component
+- Dynamically provide user feedback as they complete a form with validations
 - Handle the event of a form submission in a component
-- Dynamically provide user feedback as they complete a form
 
 ### Overview
 Since we've already learn how event handling works in React, we should be able to take the application of this knowledge one step further to allow us to submit a set of form data. Forms work a bit different than other HTML elements because they maintain some of their own state.
 
 Additionally, one of the most powerful aspects of using any JavaScript framework is to provide some dynamic user interaction. We'll see here how to integrate ongoing user feedback, so the user does not have to wait until the form is submitted to know that they might have issues with the data.
 
+## A Regular HTML Form
+
+A normal HTML form to add a student to a list might look like this:
+
+```html
+<form id="new-student" >
+  <label for="name">Student Name</label>
+  <input name="name" type="text">
+  <label for="email">Student Email</label>
+  <input name="email" type="text">
+  <input type="submit" value="Add Student">
+</form>
+```
+
+When a user enters text in the form controls the DOM stores the values of the form fields.  However we can do a lot more combining it with React.
+
+First we will build a React component to manage creating new students.
+
+```javascript
+import React, { Component } from 'react';
+import  './new_student_form.css';
+
+class NewStudentForm extends Component {
+  constructor() {
+    super();
+  }
```

Comment:
Do we need to define a constructor in this if all it does is call `super()`? This is an earnest question; I don't know React well enough